### PR TITLE
maint: rename `join-pairs` to `merge-pairs`

### LINKED ISSUE #84: Rename `join-pairs` to `merge-pairs`
**Suggestion:** Rename the q2-vsearch `join-pairs` action to `merge_pairs`, and update the associated descriptions to state something like: "Merge paired-end sequence reads..."). Merging reads is not the same as joining reads. 

**Reason:** Calling this QIIME 2 action `join-pairs` may be confusing for savvy vsearch users, as there is in fact a `--fastq_join` command within vsearch. This command will stitch each read together end-to-end with a padding string, usually Ns. As we're actually calling the vsearch `--fastq_mergepairs`, we should update the name of the action.

### PR DESCRIPTION
**Pull request by colinbrislawn**
Addresses #84

Ready for review, all feedback welcome!

EDIT: @lizgehret, here's the Description for this PR:

Description: Rename the q2-vsearch `--join_pairs` action to `--merge_pairs` so it matches the vsearch command. The output semantic type is still `JoinedSequencesWithQuality` to preserve compatibility with other plugins.

### CODE REVIEW

File: q2_vsearch/plugin_setup.py [hunk 1]
```diff
@@ -283,11 +283,11 @@
             0, 8, inclusive_start=True, inclusive_end=True)
     },
     outputs=[
-        ('joined_sequences', SampleData[JoinedSequencesWithQuality])
+        ('merged_sequences', SampleData[JoinedSequencesWithQuality])
```
Comment (colinbrislawn) [inline q2_vsearch/plugin_setup.py:286]: See discussion of the name of this semantic type.
https://github.com/qiime2/q2-vsearch/issues/84#issuecomment-1249606494

File: q2_vsearch/plugin_setup.py [hunk 2]
```diff
@@ -296,25 +296,25 @@
...
+        'minmergelen': ('Minimum length of the merging read to be retained.'),
+        'maxmergelen': ('Maximum length of the merging read to be retained.'),
+        'maxee': ('Maximum number of expected errors in the merging read '
                   'to be retained.'),
         'threads': ('The number of threads to use for computation. Does '
                     'not scale much past 4 threads.')
     },
     output_descriptions={
-        'joined_sequences': ('The joined sequences.'),
+        'merged_sequences': ('The merging sequences.'),
```
Comment (colinbrislawn) [inline q2_vsearch/plugin_setup.py]: Spelling, lol

**Comment by colinbrislawn** [comment]
@lizgehret, would you like to include this in 2022.11 or wait for now?
The full discussion around this PR is here: https://github.com/qiime2/q2-vsearch/issues/84

File: q2_vsearch/plugin_setup.py [hunk 3]
```diff
@@ -296,25 +296,25 @@
                    'discarded.'),
         'maxns': ('Sequences with more than maxns N characters are '
                   'discarded.'),
-        'allowmergestagger': ('Allow joining of staggered read pairs.'),
+        'allowmergestagger': ('Allow mergeing of staggered read pairs.'),
```
Comment (lizgehret) [inline q2_vsearch/plugin_setup.py]:
```suggestion
        'allowmergestagger': ('Allow merging of staggered read pairs.'),
```

File: q2_vsearch/plugin_setup.py [hunk 4]
```diff
@@ -296,25 +296,25 @@
...
         'minovlen': ('Minimum overlap length of forward and reverse reads '
-                     'for joining.'),
+                     'for merging.'),
         'maxdiffs': ('Maximum number of mismatches in the forward/reverse '
-                     'read overlap for joining.'),
-        'minmergelen': ('Minimum length of the joined read to be retained.'),
-        'maxmergelen': ('Maximum length of the joined read to be retained.'),
-        'maxee': ('Maximum number of expected errors in the joined read '
+                     'read overlap for merging.'),
+        'minmergelen': ('Minimum length of the merging read to be retained.'),
```
Comment (lizgehret) [inline q2_vsearch/plugin_setup.py]:
```suggestion
        'minmergelen': ('Minimum length of the merged read to be retained.'),
```

File: q2_vsearch/plugin_setup.py [hunk 5]
```diff
@@ -296,25 +296,25 @@
...
+        'maxmergelen': ('Maximum length of the merging read to be retained.'),
```
Comment (lizgehret) [inline q2_vsearch/plugin_setup.py]:
```suggestion
        'maxmergelen': ('Maximum length of the merged read to be retained.'),
```

File: q2_vsearch/plugin_setup.py [hunk 6]
```diff
@@ -296,25 +296,25 @@
...
+        'maxee': ('Maximum number of expected errors in the merging read '
```
Comment (lizgehret) [inline q2_vsearch/plugin_setup.py]:
```suggestion
        'maxee': ('Maximum number of expected errors in the merged read '
```

File: q2_vsearch/plugin_setup.py (hunk 2 above)
Comment (lizgehret) [inline q2_vsearch/plugin_setup.py]:
```suggestion
        'merged_sequences': ('The merged sequences.'),
```

**Review by lizgehret — COMMENTED** [review]
This all looks good @colinbrislawn - we should definitely be able to add this into the 2022.11 release. I'll pull this down locally to make sure everything plays nicely, but aside from the few syntax adjustments (inline) this should be good to go!

**Review by lizgehret — APPROVED** [review]
lgtm - pulled down and tested locally and everything looks reasonable!

**Pull request merged**